kytos/core/tcp_server.py

```diff
@@ -44,113 +40,108 @@ def __init__(self, server_address, RequestHandlerClass,
         """
         super().__init__(server_address, RequestHandlerClass,
                          bind_and_activate=False)
-        # Registering the register_event 'endpoint' on the server to be
-        #   accessed on the RequestHandler
         self.controller = controller
-        # self.controller_put_raw_event = controller_put_raw_event
 
     def serve_forever(self, poll_interval=0.5):
-        """Handle requests until an explicit shutdown() request."""
+        """Handle requests until an explicit shutdown() is called."""
+        SOCKET_BIND_WAIT_TIME = 0.5
+        SOCKET_BIND_N_TRIES = 2
+        # Tries to bind the socket n times, it may still be
+        # in WAIT state from previous connection.
+        for i in range(SOCKET_BIND_N_TRIES):
```

Comment:
Hum.... I still not sure if what way to go.
@beraldoleal ? @cemsbr ? @renanrodrigo ? @abaruchi ?
Any thoughts?
Just adding one more link:
http://www.unixguide.net/network/socketfaq/4.5.shtml

Just to clarify for those that are just getting into the discussion.

After a socket is closed, it's state is changed to "closed" but it remains opened for some time (`TIME_WAIT`). This is a default behavior from Unix sockets.
If anyone tries to establish a new socket on the same "socket"*, then the OS will raise `Address already in use` error.

We have three options here.
1) Just raise the error;
2) Use a `timeout` and keep trying to bind to the socket again. [ @erickvermot proposal on this PR ];
3) Use the `socket.SO_REUSEADDR` to tell the OS that it should get the living (but closed) socket and "restart" it.

*A socket is a 5 tuple (`proto`, `local addr`, `local port`, `remote addr`, `remote port`)